### PR TITLE
Setup AWS S3 Data Validation

### PR DESCRIPTION
**Pull request by daniel-ji**
Setup the folder structure of AWS S3 data validation scripts, just adding Pytest fixtures with the folder structure. Tests and Allure reporting will be added in later frameworks. 

A large majority of the Pytest code written is from Utz's scripts in the cryoet-portal repo, so thank you @uermel for all of this!

### CODE REVIEW

**Review by jgadling — APPROVED** [review]

File: ingestion_tools/scripts/data_validation/fixtures/data.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+@pytest.fixture(scope="session")
+def tiltseries_metadata(tiltseries_meta_file: str, filesystem: FileSystemApi) -> Dict:
+    """Load the tiltseries metadata."""
+    with filesystem.open(tiltseries_meta_file, "r") as f:
+        return json.load(f)
+
+
+@pytest.fixture()
+def tiltseries_mdoc(tiltseries_mdoc_file: str, filesystem: FileSystemApi) -> pd.DataFrame:
```
Comment (manasaV3) [inline ingestion_tools/scripts/data_validation/fixtures/data.py:40]: Curious why is this not setup for "session" scope?

File: ingestion_tools/scripts/data_validation/fixtures/data.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        try:
+            headers[file] = get_header(file)
+        except Exception as _:
+            pytest.fail(f"Failed to get header for {file}")
+
+    return headers
+
+
+@pytest.fixture(scope="session")
+def tiltseries_metadata(tiltseries_meta_file: str, filesystem: FileSystemApi) -> Dict:
```
Comment (manasaV3) [inline ingestion_tools/scripts/data_validation/fixtures/data.py:33]: Could this just be metadata and shared across all metadata reads?

File: ingestion_tools/scripts/data_validation/fixtures/data.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+def tiltseries_tlt(tiltseries_tlt_file: str, filesystem: FileSystemApi) -> pd.DataFrame:
+    """Load the tiltseries tlt."""
+
+    with filesystem.open(tiltseries_tlt_file, "r") as f:
+        return pd.read_csv(f, sep=r"\s+", header=None, names=["tilt_angle"])
+
+
+### Tomogram fixtures ###
+@pytest.fixture(scope="session")
+def canonical_tomo_mrc_headers(canonical_tomo_mrc_files: List[str]) -> Dict[str, MrcInterpreter]:
```
Comment (manasaV3) [inline ingestion_tools/scripts/data_validation/fixtures/data.py:57]: Could this also be shared with the `tiltseries_mrc_headers`?

File: ingestion_tools/scripts/data_validation/fixtures/data.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+@pytest.fixture(scope="session")
+def canonical_tomogram_metadata(canonical_tomo_meta_file: str, filesystem: FileSystemApi) -> Dict:
+    """Load the canonical tomogram metadata."""
+    with filesystem.open(canonical_tomo_meta_file, "r") as f:
+        return json.load(f)
+
+
+### Annotation fixtures ###
+@pytest.fixture(scope="session")
+def annotation_metadata(annotation_meta_files: str, filesystem: FileSystemApi) -> Dict[str, Dict]:
```
Comment (manasaV3) [inline ingestion_tools/scripts/data_validation/fixtures/data.py:78]: Should `annotation_meta_files` be a list[str]?

File: ingestion_tools/scripts/data_validation/fixtures/data.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                annotations[annoname][file] = ndjson.load(f)
+
+    return annotations
+
+
+@pytest.fixture(scope="session")
+def oriented_point_annotations(
+    oriented_point_annotation_files: Dict[str, List[str]],
+    filesystem: FileSystemApi,
+) -> Dict[str, Dict[str, Dict]]:
```
Comment (manasaV3) [inline ingestion_tools/scripts/data_validation/fixtures/data.py:109]: Can this and the above method be refactored into 1 method?

File: ingestion_tools/yarn.lock [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+allure@^0.0.0:
```
Comment (manasaV3) [inline ingestion_tools/yarn.lock:5]: I believe we wanted to not have this committed in for now.

Comment (daniel-ji) [reply ingestion_tools/yarn.lock:5]: Sorry, removed!

File: ingestion_tools/scripts/data_validation/fixtures/data.py (hunk 2 above)
Comment (daniel-ji) [reply ingestion_tools/scripts/data_validation/fixtures/data.py:33]: tiltseries_metadata.json is different than tomogram_metadata.json and they do not share the same information that can be used by both tiltseries validation tests and tomogram validation tests? dataset_metadata.json doesn't have the information that tiltseries validation tests and tomogrma validation tests need either.

File: ingestion_tools/scripts/data_validation/fixtures/data.py (hunk 3 above)
Comment (daniel-ji) [reply ingestion_tools/scripts/data_validation/fixtures/data.py:69]: refactored!

File: ingestion_tools/scripts/data_validation/fixtures/data.py (hunk 1 above)
Comment (daniel-ji) [reply ingestion_tools/scripts/data_validation/fixtures/data.py:52]: I believe it should be, changed!

File: ingestion_tools/scripts/data_validation/fixtures/data.py (hunk 5 above)
Comment (daniel-ji) [reply ingestion_tools/scripts/data_validation/fixtures/data.py:119]: Done!

File: ingestion_tools/scripts/data_validation/fixtures/data.py (hunk 4 above)
Comment (daniel-ji) [reply ingestion_tools/scripts/data_validation/fixtures/data.py:83]: fixed!

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**